Nomad 1.7.0-beta.1 crashes on Raspberry Pi 3B running Raspbian 10

### Nomad version
1.7.0-beta.1

### Operating system and Environment details
Raspberry Pi 3 Model B Rev 1.2
Raspbian 10

### Issue
Nomad segfaults on startup

### Reproduction steps
Running `nomad agent -dev` 

#### Expected Result
Nomad successfully starts up in dev mode


#### Actual Result
```
# ./nomad agent -dev
==> No configuration files loaded
==> Starting Nomad agent...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbadb28]

goroutine 1 [running]:
github.com/hashicorp/nomad/client/lib/idset.(*Set[...]).Size(...)
        github.com/hashicorp/nomad/client/lib/idset/idset.go:190
github.com/hashicorp/nomad/client/lib/numalib.(*Sysfs).discoverCosts(0x4919c28, 0x68a6780)
        github.com/hashicorp/nomad/client/lib/numalib/detect_linux.go:70 +0x18
github.com/hashicorp/nomad/client/lib/numalib.(*Sysfs).ScanSystem(0x4919c28, 0x68a6780)
        github.com/hashicorp/nomad/client/lib/numalib/detect_linux.go:52 +0x98
github.com/hashicorp/nomad/client/lib/numalib.Scan(...)
        github.com/hashicorp/nomad/client/lib/numalib/detect.go:23
github.com/hashicorp/nomad/client/fingerprint.(*CPUFingerprint).initialize(0x6e8e0d0, 0x70157c0)
        github.com/hashicorp/nomad/client/fingerprint/cpu.go:90 +0x480
github.com/hashicorp/nomad/client/fingerprint.(*CPUFingerprint).Fingerprint(0x6e8e0d0, 0x70157c0, 0x70126a8)
        github.com/hashicorp/nomad/client/fingerprint/cpu.go:40 +0x24
github.com/hashicorp/nomad/client.(*FingerprintManager).fingerprint(0x711b6b0, {0x2ab6ced, 0x3}, {0x32359a8, 0x6e8e0d0})
        github.com/hashicorp/nomad/client/fingerprint_manager.go:200 +0xd8
github.com/hashicorp/nomad/client.(*FingerprintManager).setupFingerprinters(0x711b6b0, {0x6f88d00, 0x13, 0x20})
        github.com/hashicorp/nomad/client/fingerprint_manager.go:143 +0xdc
github.com/hashicorp/nomad/client.(*FingerprintManager).Run(0x711b6b0)
        github.com/hashicorp/nomad/client/fingerprint_manager.go:107 +0x4c0
github.com/hashicorp/nomad/client.NewClient(0x6a87c80, {0x3233fb0, 0x6832128}, 0x6832160, {0x323f0e4, 0x6e88080}, 0x0)
        github.com/hashicorp/nomad/client/client.go:467 +0x10cc
github.com/hashicorp/nomad/command/agent.(*Agent).setupClient(0x6844240)
        github.com/hashicorp/nomad/command/agent/agent.go:1124 +0x304
github.com/hashicorp/nomad/command/agent.NewAgent(0x68d7560, {0x324da2c, 0x6db32a8}, {0x322a21c, 0x6ddc630}, 0x6df9880)
        github.com/hashicorp/nomad/command/agent/agent.go:160 +0x218
github.com/hashicorp/nomad/command/agent.(*Command).setupAgent(0x6d8f200, 0x68d7560, {0x324da2c, 0x6db32a8}, {0x322a21c, 0x6ddc630}, 0x6df9880)
        github.com/hashicorp/nomad/command/agent/command.go:597 +0x70
github.com/hashicorp/nomad/command/agent.(*Command).Run(0x6d8f200, {0x68be058, 0x1, 0x1})
        github.com/hashicorp/nomad/command/agent/command.go:810 +0x644
github.com/mitchellh/cli.(*CLI).Run(0x69e98c0)
        github.com/mitchellh/cli@v1.1.5/cli.go:262 +0x4b4
main.Run({0x68be050, 0x2, 0x2})
        github.com/hashicorp/nomad/main.go:110 +0x230
main.main()
        github.com/hashicorp/nomad/main.go:80 +0x58
```

--- a/client/lib/numalib/detect_linux.go
+++ b/client/lib/numalib/detect_linux.go
@@ -67,6 +67,10 @@ func (*Sysfs) discoverOnline(st *Topology) {
 }
 
 func (*Sysfs) discoverCosts(st *Topology) {
+	if st.NodeIDs.Empty() {
+		return
+	}
+
 	dimension := st.NodeIDs.Size()
 	st.Distances = make(SLIT, st.NodeIDs.Size())
 	for i := 0; i < dimension; i++ {
@@ -94,24 +98,39 @@ func (*Sysfs) discoverCores(st *Topology) {
 	}
 	st.Cores = make([]Core, onlineCores.Size())
 
-	_ = st.NodeIDs.ForEach(func(node hw.NodeID) error {
-		s, err := os.ReadFile(fmt.Sprintf(cpulistFile, node))
-		if err != nil {
-			return err
-		}
-
-		cores := idset.Parse[hw.CoreID](string(s))
-		_ = cores.ForEach(func(core hw.CoreID) error {
-			// best effort, zero values are defaults
-			socket, _ := getNumeric[hw.SocketID](cpuSocketFile, core)
+	switch {
+	case st.NodeIDs == nil:
+		// We did not find node data, no node to associate with
+		_ = onlineCores.ForEach(func(core hw.CoreID) error {
+			st.NodeIDs = idset.From[hw.NodeID]([]hw.NodeID{0})
+			const node = 0
+			const socket = 0
 			max, _ := getNumeric[hw.KHz](cpuMaxFile, core)
 			base, _ := getNumeric[hw.KHz](cpuBaseFile, core)
-			siblings, _ := getIDSet[hw.CoreID](cpuSiblingFile, core)
-			st.insert(node, socket, core, gradeOf(siblings), max, base)
+			st.insert(node, socket, core, Performance, max, base)
 			return nil
 		})
-		return nil
-	})
+	default:
+		// We found node data, associate cores to nodes
+		_ = st.NodeIDs.ForEach(func(node hw.NodeID) error {
+			s, err := os.ReadFile(fmt.Sprintf(cpulistFile, node))
+			if err != nil {
+				return err
+			}
+
+			cores := idset.Parse[hw.CoreID](string(s))
+			_ = cores.ForEach(func(core hw.CoreID) error {
+				// best effort, zero values are defaults
+				socket, _ := getNumeric[hw.SocketID](cpuSocketFile, core)
+				max, _ := getNumeric[hw.KHz](cpuMaxFile, core)
+				base, _ := getNumeric[hw.KHz](cpuBaseFile, core)
+				siblings, _ := getIDSet[hw.CoreID](cpuSiblingFile, core)
+				st.insert(node, socket, core, gradeOf(siblings), max, base)
+				return nil
+			})
+			return nil
+		})
+	}
 }
 
 func getIDSet[T idset.ID](path string, args ...any) (*idset.Set[T], error) {

